Homework-4-ManyToManyRelationships/src/api/users.ts

```diff
@@ -0,0 +1,102 @@
+import express, { Request, Response, NextFunction } from 'express';
+import {
+  ValidatedRequest,
+  createValidator
+} from 'express-joi-validation';
+import { Container } from "typedi";
+import { IRequestBodySchema, reqBodySchema } from "../schemas/userSchema";
+import { UserService } from "../services/userService";
+import { BaseUserDTO, UserDTO } from "../types/userDTO";
+import { UserFilter } from "../types/filters";
+
+export const usersRouter = express.Router();
+const validator = createValidator({ passError: true });
+const userService = Container.get(UserService);
+
+const getUsers = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const filters: UserFilter = req.query;
+    const users = await userService.getUsers(filters);
+    res.status(200).send(users);
+  } catch (error: any) {
+    next(error);
+  }
+};
+
+const getUserById = async (req: Request, res: Response, next:NextFunction) => {
```

Comment:
There is quite a lot of common things between api/groups and api/users. Probably not a problem with two entities. But you can have way more.

You can check `feathers` for inspiration.
The same idea could be applied to services.

Reply:
Thanks for advice